core/src/main/java/com/gentics/mesh/core/data/node/impl/NodeImpl.java

```diff
@@ -1385,11 +1395,67 @@ public void delete(BulkActionContext bac, boolean ignoreChecks, boolean recursiv
 			log.debug("Deleting node {" + getUuid() + "} vertex.");
 		}
 
-		bac.add(onDeleted(getUuid(), null, getSchemaContainer(), null, null, null));
+		addReferenceUpdates(bac);
+
+		bac.add(onDeleted(getUuid(), getSchemaContainer(), null, null, null));
 		getElement().remove();
 		bac.process();
 	}
 
+	private void addReferenceUpdates(BulkActionContext bac) {
+		Set<String> handledNodeUuids = new HashSet<>();
+
+		// Check whether NodeGraphFields (in either NGFC, or Micronode) reference this node.
+		for (NodeGraphField field : inE(HAS_FIELD).has(NodeGraphFieldImpl.class).frameExplicit(NodeGraphFieldImpl.class)) {
+			for (GraphFieldContainer container : field.outV().frame(GraphFieldContainer.class)) {
+				followContainer(bac, container, handledNodeUuids);
+			}
+		}
+
+		// Check whether Node lists (in either NGFC, or Micronode) reference this node.
+		for (NodeGraphField listItem : inE(HAS_ITEM).has(NodeGraphFieldImpl.class).frameExplicit(NodeGraphFieldImpl.class)) {
```

Comment:
We don't need to inspect the edge, use `in` instead

Reply:
Not possible since we need to do a type check on the edge